### PR TITLE
Reports - Update Location wise Registration report

### PR DESCRIPTION
**Pull request by riyaTw**

### CODE REVIEW

File: openmrs/apps/reports/sql/registeredPatients.sql [hunk 1]
```diff
@@ -4,21 +4,24 @@ SELECT
...
-  LEFT OUTER JOIN (SELECT ei.patient_id, ei.identifier 
-    FROM patient_identifier ei 
+		ON pt.patient_id = primaryIdentifier.patient_id
+  LEFT OUTER JOIN (SELECT ei.patient_id, ei.identifier
+    FROM patient_identifier ei
       JOIN patient_identifier_type pit ON ei.identifier_type = pit.patient_identifier_type_id AND pit.retired is FALSE AND ei.preferred = 1
       JOIN global_property gp ON gp.property="bahmni.extraPatientIdentifierTypes" AND INSTR (gp.property_value, pit.uuid)) extraIdentifier 
 		ON pt.patient_id = extraIdentifier.patient_id
   CROSS JOIN (SELECT @rownum := 0) AS dummy
+  LEFT JOIN encounter e ON pt.patient_id = e.patient_id AND e.encounter_type = 2
```
Comment (mohan-13) [inline openmrs/apps/reports/sql/registeredPatients.sql]: What would happen if the sync happens from Avni ? Also having 2 hardcoded here might break the report if the id chnages, so better to verify with Name

**Comment by mohan-13** [comment]
Is the encounter getting created when I just hit on Save in the registration page ?

**Comment by riyaTw** [comment]
> Is the encounter getting created when I just hit on Save in the registration page ?

No it is not getting created

File: openmrs/apps/reports/sql/registeredPatients.sql (hunk 1 above)
Comment (riyaTw) [reply openmrs/apps/reports/sql/registeredPatients.sql]: For patients registered from Avni also "Location Name" column is filled correctly on the report. Removed the hard cording for 2

File: openmrs/apps/reports/sql/registeredPatients.sql [hunk 2]
```diff
@@ -4,21 +4,25 @@ SELECT
...
+                 FROM patient_identifier ei
+                 JOIN patient_identifier_type pit ON ei.identifier_type = pit.patient_identifier_type_id AND pit.retired is FALSE AND ei.preferred = 1
+                 JOIN global_property gp ON gp.property="bahmni.extraPatientIdentifierTypes" AND INSTR (gp.property_value, pit.uuid)) extraIdentifier 
 		ON pt.patient_id = extraIdentifier.patient_id
-  CROSS JOIN (SELECT @rownum := 0) AS dummy
-  WHERE pt.voided is FALSE
-  AND cast(CONVERT_TZ(pt.date_created,'+00:00','+5:30') AS DATE) BETWEEN '#startDate#' AND '#endDate#';
\ No newline at end of file
+CROSS JOIN (SELECT @rownum := 0) AS dummy
+LEFT JOIN encounter e ON e.patient_id = pt.patient_id
```
Comment (mohan-13) [inline openmrs/apps/reports/sql/registeredPatients.sql:24]: Shouldn't we filter only REG and Community Registration encounters? There could be different encounters for a patient

Comment (riyaTw) [reply openmrs/apps/reports/sql/registeredPatients.sql:24]: Yes we did that but for those patients who have just been saved in the system those will not show up in the reports if we do this filter that's why we removed it

File: openmrs/apps/reports/sql/registeredPatients.sql [hunk 3]
```diff
@@ -4,21 +4,25 @@ SELECT
...
-  CROSS JOIN (SELECT @rownum := 0) AS dummy
-  WHERE pt.voided is FALSE
-  AND cast(CONVERT_TZ(pt.date_created,'+00:00','+5:30') AS DATE) BETWEEN '#startDate#' AND '#endDate#';
\ No newline at end of file
+CROSS JOIN (SELECT @rownum := 0) AS dummy
+LEFT JOIN encounter e ON e.patient_id = pt.patient_id 
+LEFT JOIN location loc ON e.location_id = loc.location_id
+WHERE pt.voided is FALSE
+AND cast(CONVERT_TZ(pt.date_created,'+00:00','+5:30') AS DATE) BETWEEN '#startDate#' AND '#endDate#'
+GROUP BY pt.patient_id;
```
Comment (angshu) [inline openmrs/apps/reports/sql/registeredPatients.sql:28]: Don't you want to order by date?

File: openmrs/apps/reports/sql/registeredPatients.sql [hunk 4]
```diff
@@ -4,21 +4,25 @@ SELECT
   concat(pn.given_name, " ", ifnull(pn.family_name, ""))                                                                                  AS "Patient Name",
   floor(DATEDIFF(NOW(), p.birthdate) / 365)                                                                                               AS "Age",
   p.gender                                                                                                                                AS "Gender",
-  DATE_FORMAT(CONVERT_TZ(pt.date_created,'+00:00','+5:30'), "%d-%b-%Y")                                                                   AS "Registration Date"
+  DATE_FORMAT(CONVERT_TZ(pt.date_created,'+00:00','+5:30'), "%d-%b-%Y")                                                                   AS "Registration Date",
```
Comment (angshu) [inline openmrs/apps/reports/sql/registeredPatients.sql:7]: Why patient.date_created as registration date? it should be encounter date. 
What happens for a returning patient - that should also count as registration.

**Comment by angshu** [comment]
What are you trying to find out? new patient created by dates? or footfall count at the location by dates? Patient.date_created = date of creation of the patient. Whereas if its meant for visit registration - then even the returning patients reg encounter date should be considered. So if a patient was registered 3 days back and returned today - the patient should list in today's regstration date as well?
Its ok if you dont want to filter by encounter type - but if your intention is for later - then you should select encounter date. also group and sort by date - since it takes a date range

**Pull request merged**